### ISSUE TITLE
Filebrowser creates root:root files in docker container

### ISSUE THREAD

**Issue by AmadeusGraves** [filebrowser/filebrowser#2053]
There is a problem with filebrowser, when you instantiate the project in a docker container, filebrowser takes by default the root user since by default the filebrowser dockerfile does not instantiate by default the user with PUID:GUID 1000:1000.

This is a problem since this causes filebrowser to always use the root user.

I leave also a url that explains in more detail the error.
https://jtreminio.com/blog/running-docker-containers-as-current-host-user/
(The solution explained in the link does not work either.)

You should change the dockerfile so that when you enter some environment parameters like PUID or GUID, it instantiates the user.

I have tried creating the container manually. That is, as it comes in the filebrowser installation guide.

`sudo docker run \
    -v /:/srv \
    -v /DATA/AppData/FileBrowser/db:/db \
    -e PUID=$(id -u) \
    -e PGID=$(id -g) \
    -u $(id -u):$(id -g) \
    -p 45123:80 \
    filebrowser/filebrowser`

I have seen from previous bugs that it is recommended to use this ` -u $(id -u):$(id -g) \`  when you instantiate the container. 
It doesn't work, because you don't have permissions to open the database.db, I think this is something to modify in the dockerfile in order to make it work when you start the container with the image.

I have also tried to insert the user 1000:1000 from the container itself by executing.
`sudo docker exec -t -i FileBrowser /bin/ash` and I have managed to insert the user but it doesn't work either.

In short, I think you need to modify the dockerfile in order to be able to create the user when the container is instantiated.
Best regards. I will also be investigating on my own to see if there is any other possibility.

**Comment by o1egl** [comment]
You can find the docker run command in the [docs](https://filebrowser.org/installation#docker)

**Comment by AmadeusGraves** [comment]
Yes, I have seen the documentation, in fact the command that I have passed in the description is based on that command, I put in the parameter that says the documentation to solve the problem. `-u $(id -u):$(id -g) \` . It does not work, because you don't have permissions to open the database.db, I think this is something to modify in the dockerfile in order to make it work when you start the container with the image.

**Comment by o1egl** [comment]
You need to fix your database file permissions. I can confirm that it works with no issues

```
$ touch filebrowser.db
$ docker run -v $(pwd)/filebrowser.db:/database.db -u $(id -u):$(id -g) -p 8081:80  filebrowser/filebrowser
```

**Comment by pjosuah** [comment]
I'd like to re-open this issue.

FileBrowser doesn't seem to take into account the PUID:GUID passed in the compose file or docker command. It still creates files as root:root. Whic is a bit annoying.

Any idea, anyone?

**Comment by unlair** [comment]
> I'd like to re-open this issue.
> 
> FileBrowser doesn't seem to take into account the PUID:GUID passed in the compose file or docker command. It still creates files as root:root. Whic is a bit annoying.
> 
> Any idea, anyone?

This worked for me using Immich, might also work for FileBrowser but haven't tried it yet:

https://www.reddit.com/r/immich/comments/16nfjab/comment/k68ydrl/

**Comment by EvandroMaddes** [comment]
@princejosuah  I overcome this issue using a simple bash job. It executes a script where the owner of each file and/or folder created on filebrowser is changed to any user of your choice. It is based on [inotify-tools](https://github.com/inotify-tools/inotify-tools/wiki)

```bash
#!/bin/bash

dir_under_control=/path/to/directory/to/monitor
outfile_permissions_logging=logging-file.log
new_user=new-owner-of-file

# pipe symbol to ensure the output of inotifywait is properly fed into the while loop
sudo inotifywait -mr --timefmt '%d/%m/%y %H:%M' --format '%T %w %f' --event create ${dir_under_control} |
while read -r date time dir file; do
  changed_abs="${dir}${file}"
  sudo chown ${new_user} "$changed_abs" >> ${outfile_permissions_logging}
  echo "[${date} ${time}] ${new_user} user new owner of file: ${changed_abs}" >> ${outfile_permissions_logging}
done
```
